### PR TITLE
test(e2e): onboarding tests

### LINKED ISSUE #163: Add e2e tests for onboarding flows
Add e2e tests to cover the onboarding flows (welcome screens)

### PR DESCRIPTION
**Pull request by QZera**
Prerequisite #169 

Closes #163 

- Tests the welcome screen shown if the rad binary isn't found (e.g. before Radicle is installed)
  - Radicle not being installed is simulated by moving the `.radicle` from it's default location
- Removes `browser.pause` (used on some flaky tests) in favour of using `browser.waitUntil` (asynchronous assertions). This should reduce overall flakiness when adding new tests to the suite.
- Extracts some common helper functions to separate files

### CODE REVIEW

File: test/e2e/helpers/assertions.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { browser } from '@wdio/globals'
+import type { Workbench } from 'wdio-vscode-service'
+
+/**
+ * Asserts that the CLI Commands and Patches sections are visible in the sidebar. This is
+ * considered the default state when the workspace is open with a git and rad initialized
+ * repository.
+ */
+export async function expectCliCommandsAndPatchesToBeVisible(workbench: Workbench) {
```
Comment (QZera) [inline test/e2e/helpers/assertions.ts:9]: I was thinking of having a more semantic name for this function. Something that expresses that it is the default view of the extension in the sidebar (e.g. `expectDefaultSidebarViewToBeVisible`).
That way, if sections are added, removed in the future, this name remains relevant. Thoughts?

Comment (maninak) [reply test/e2e/helpers/assertions.ts:9]: Good point. While both work, I think the alternative more semantic name would better match how we use the func and scales better over time.

Though it may be more accurate to use a name like `expectStandardSidebarViewsToBeVisible`.

Comment (QZera) [reply test/e2e/helpers/assertions.ts]: Renamed as suggested

File: test/e2e/specs/onboarding.spec.ts [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+
+    before(async () => {
+      // Simulate radicle "not being installed" by renaming the node home directory
+      await $`mv ${nodeHomePath} ${tempPathToNodeHome}`
+    })
+
+    after(async () => {
+      await $`mv ${tempPathToNodeHome} ${nodeHomePath}`
+      await workbench.executeCommand('Developer: Reload Window')
+    })
```
Comment (maninak) [inline test/e2e/specs/onboarding.spec.ts:30]: It seems that (all things considered) we are

1. booting VS Code
2. activating our extension
3. "deleting" the home folder
4. testing
5. "un-deleting" the hove folder

TL;DR: Step 3 must happen before Step 2.

While of course having this is better than not at all, I'm pretty sure that to properly emulate the onboarding flow, the home folder must not exist _before activating the extension_. That is the trickiest part of the detection logic, as we also saw in our code walkthrough, and I'm afraid this approach isn't covering it. And that's arguably the single most critical part of the onboarding flow.

**Review by maninak — CHANGES_REQUESTED** [review]
Please let me know if the single comment-assumption I've made is inaccurate and I'll insta-approve.

In general, and from a reader's perspective, I like how the testing architecture is shaping up, well done!

Naturally, I'll be able to form a more thorough verdict after I write a few tests of my own.

File: test/e2e/specs/onboarding.spec.ts (hunk 2 above)
Comment (QZera) [reply test/e2e/specs/onboarding.spec.ts]: Great point! You have the order right.

I've moved the "deletion" of the home directory before Step 1 (AFAIK we don't have a good way to do things between Step 1 and Step 2).

There's a small caveat. I had to use wdio's `onWorkerStart` [hook](https://webdriver.io/docs/configuration/#onworkerstart), which is defined in `wdio.conf.ts`. Ideally I would 've liked this logic to be located in the suite itself itself but alas, I don't think there's much we can do about that.

Comment (maninak) [reply test/e2e/specs/onboarding.spec.ts]: Good! The hook solution is actually not as bad as I thought it could have been, well done finding it!

I won't be surprised if you come up with an even better approach later too, probably will be something more "out of the box", as things are.

**Review by maninak — APPROVED** [review]
Well done, zac, and thank you!
